ts/src/models/transaction_model.ts

```diff
@@ -44,7 +45,13 @@ export const transactionModel = {
             .leftJoinAndSelect('transaction.takerAssetFillAmounts', 'takerAssetFillAmount')
             .where('order.hash IN (:...orderHashes)', { orderHashes });
         if (opts !== undefined && opts.takerAddress !== undefined) {
```

Comment:
We actually want `takerAddress` to be undefined here. Basically if `takerAddress !== undefined`, query by `takerAddress`. If `txOrigin !== undefined`, query by `txOrigin`.

We could keep that logic all in this function, but I think the intent might be more clear (and more extensible) if we just broke this up into 2 functions.

Reply:
I believe I've changed the query logic to match what you've described (here, and in your other comment).

But, I didn't break apart into 2 functions as you suggested.  Given that the query by `txOrigin` occurs in two slightly different situations, it wasn't clear to me where exactly to break things apart, but if you can advise further, I'm happy to do it.